src/main/java/edu/iu/terracotta/service/app/impl/QuestionSubmissionServiceImpl.java

```diff
@@ -336,22 +336,11 @@ public void validateQuestionSubmission(QuestionSubmissionDto questionSubmissionD
     }
 
     @Override
-    public void canSubmit(String canvasCourseId, String assignmentId, String canvasUserId, long deploymentId) throws CanvasApiException, AssignmentAttemptException, IOException {
-        // We find the right deployment:
-        Optional<ToolDeployment> toolDeployment = allRepositories.toolDeploymentRepository.findById(deploymentId);
+    public void canSubmit(String canvasCourseId, String assignmentId, String canvasUserId, long platformDeploymentId)
+            throws CanvasApiException, AssignmentAttemptException, IOException {
 
-        if (!toolDeployment.isPresent()) {
-            return;
-        }
-
-        String litDeploymentId = toolDeployment.get().getLtiDeploymentId();
-        List<PlatformDeployment> platformDeploymentList = allRepositories.platformDeploymentRepository.findByToolDeployments_LtiDeploymentId(litDeploymentId);
-
-        if (platformDeploymentList.isEmpty()) {
-            return;
-        }
-
-        PlatformDeployment platformDeployment = platformDeploymentList.get(0);
+        PlatformDeployment platformDeployment = allRepositories.platformDeploymentRepository
+                .getOne(platformDeploymentId);
```

Comment:
I know this is after the merge, but getOne() is deprecated in later Spring versions.

Reply:
Good to know, thanks @relong76 . Is there a non deprecated method that does the equivalent of "findById" but doesn't return an Optional?